### PR TITLE
News-Item Persistence Class

### PR DESCRIPTION
**Pull request by umangutkarsh**
**This PR provides the funcitonality of querying the NewsItem{} into the Rocket.Chat persistence storage.**

It addresses the following:

- [X] Create a `NewsItemPersistence` class.
- [X] Define methods like `saveNews(), getNewsById(), removeNewsById(), removeAllNews()`

P.S. - Some minor error handlings are required.

### CODE REVIEW

**Comment by umangutkarsh** [comment]
Hey @abhinavkrin . Please review this one too! Thanks.

**Review by abhinavkrin — APPROVED** [review]

**Comment by umangutkarsh** [comment]
@abhinavkrin . Ready to merge.

File: news-aggregation/persistence/NewsItemPersistence.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            ),
+            new RocketChatAssociationRecord(
+                RocketChatAssociationModel.MISC,
+                news.id,
+            ),
+        ];
+
+        const newsToBeDeleted = await this.getNewsById(news);
+
+        let removedNews: object[];
```
Comment (abhinavkrin) [inline news-aggregation/persistence/NewsItemPersistence.ts:103]: This object is assigned a value but not used anywhere. Did you mean to return it ?

File: news-aggregation/persistence/NewsItemPersistence.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            console.log("News saved!!", recordId);
+        } catch (err) {
+            console.error("Could not save news in persistence.", err);
+            this.app
+                .getLogger()
+                .error("Could not save news in persistence.", err);
+        }
+    }
+
+    async getNewsById(news: NewsItem): Promise<object[]> {
```
Comment (abhinavkrin) [inline news-aggregation/persistence/NewsItemPersistence.ts:55]: Should return `Promise<NewsItem>` instead of `Promise<object>`

**Review by abhinavkrin — APPROVED** [review]
I have left some comment which you can improve later on.

**Pull request merged**